<?xml version="1.0"?>
<robot name="LeArm">
  <!-- UNITS DISTANCE(METERS) ROTATION(RADIANS) -->

	<link name="base_link"></link>
  <link name="shoulder_link"></link>
  <link name="humerus_link"></link>
  <link name="forearm_link"></link>
  <link name="wrist_link"></link>
  <link name="hand_link"></link>
  <link name="grip_left_link"></link>
  <link name="grip_right_link"></link>

  <joint name="shoulder_pan" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin xyz="0 0 0.070" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.57" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="shoulder_lift" type="revolute">
    <parent link="shoulder_link"/>
    <child link="humerus_link"/>
    <origin xyz="0 0 0.015" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="elbow" type="revolute">
    <parent link="humerus_link"/>
    <child link="forearm_link"/>
    <origin xyz="0 0 0.105" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="wrist_flex" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_link"/>
    <origin xyz="0 0 0.090" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="wrist_roll" type="revolute">
    <parent link="wrist_link"/>
    <child link="hand_link"/>
    <origin xyz="0 0 0.065" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.57" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="grip_left" type="revolute">
    <parent link="hand_link"/>
    <child link="grip_left_link"/>
    <origin xyz="0 +0.015 0.035" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
    <limit lower="-1.57" upper="0" effort="1000" velocity="0.5"/>
  </joint>
  <joint name="grip_right" type="revolute">
    <parent link="hand_link"/>
    <child link="grip_right_link"/>
    <origin xyz="0 -0.015 0.035" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
    <limit lower="0" upper="1.57" effort="1000" velocity="0.5"/>
  </joint>
</robot>

